ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -108,31 +118,51 @@ class BaseLine extends React.Component {
       return false
     }
   }
+  validatePoint(point) {
+    if (
+      point.length !== 2 ||
+      (Number.isNaN(Number.parseFloat(point[0])) &&
+        Number.isNaN(Number.parseFloat(point[1])))
+    ) {
+      return JSON.stringify(point) + ' is not a valid point.'
+    } else if (
+      point[0] > 180 ||
+      point[0] < -180 ||
+      point[1] > 90 ||
+      point[1] < -90
+    ) {
+      return JSON.stringify(point) + ' is not a valid point.'
+    }
+    return ''
+  }
   validateListOfPoints(coordinates) {
     let message = ''
-    if (this.props.mode === 'poly' && coordinates.length < 4) {
-      message = 'Minimum of 4 points needed for polygon'
-    } else if (this.props.mode === 'line' && coordinates.length < 2) {
-      message = 'Minimum of 2 points needed for line'
+    const isLine = this.props.mode.includes('line')
+    let numPoints = isLine ? 2 : 4
+    if (!this.props.mode.includes('multi')) {
+      if (coordinates.some(coords => coords.length > 2)) {
+        message = ''
```

Comment:
I put this check in here for the case when a MultiLine/MultiPolygon is pasted into the field while in Line/Polygon mode. We don't want to perform the check in the `else if` if this check is true because it wouldn't apply. My initial thought was to keep the checks separate in case we ever decide to implement MultiLine or MultiPolygon modes in queries, then we'd want a message here to switch to those modes. But I don't know if we'll ever decide to do that, so I'll negate this check and include it with the others.